README.md

```diff
@@ -3,58 +3,153 @@ xqueue_watcher
 
 This is an implementation of a polling [XQueue](https://github.com/edx/xqueue) client and grader.
 
+Overview
+========
 
-Running
-=======
-
-`python -m xqueue_watcher -d [path to settings directory]`
-
-
-JSON configuration file
-=======================
-	{
-		"test-123": {
-			"SERVER": "http://127.0.0.1:18040",
-			"CONNECTIONS": 1,
-			"AUTH": ["lms", "lms"],
-			"HANDLERS": [
-				{
-					"HANDLER": "xqueue_watcher.grader.Grader",
-					"KWARGS": {
-						"grader_root": "/path/to/course/graders/",
-					}
-				}
-			]
-		}
-	}
+There are several components in a working XQueue Watcher service:
+- **XQueue Watcher**: it polls an xqueue service continually for new submissions and grades them.
+- **Submissions Handler**: when the watcher finds any new submission, it will be passed to the handler for grading. It is a generic handler that can be configured to work with different submissions through individual submission graders.
+- **Individual Submission Grader**: each exercise or homework may specify its own "grader". This should map to a file on the server that usually specifies test cases or additional processing for the student submission.
+
+Usually your server will look like this:
+```
+root/
+├── xqueue-watcher/
+│   ├── ... # xqueue-watcher repo, unchanged
+│   └── ...
+├── config/
+│   └── conf.d/
+│   │   └── my-course.json
+│   └── logging.json
+└── my-course/
+   ├── exercise1/
+   │   ├── grader.py  # - per-exercise grader
+   │   └── answer.py  # - if using JailedGrader
+   ├── ...
+   └── exercise2/
+       ├── grader.py
+       └── answer.py
+```
+Running XQueue Watcher:
+======================
+
+Usually you can run XQueue Watcher without making any changes. You should keep course-specific in another folder like shown above, so that you can update xqueue_watcher anytime.
+
+Install the requirements before running `xqueue_watcher`
+```bash
+cd xqueue-watcher/
+make requirements
+```
+
+Now you're ready to run it.
+```bash
+python -m xqueue_watcher -d [path to the config directory, eg ../config]
+```
+
+The course configuration JSON file in `conf.d` should have the following structure:
+```json
+    {
+        "test-123": {
+            "SERVER": "http://127.0.0.1:18040",
+            "CONNECTIONS": 1,
+            "AUTH": ["lms", "lms"],
+            "HANDLERS": [
+                {
+                    "HANDLER": "xqueue_watcher.grader.Grader",
+                    "KWARGS": {
+                        "grader_root": "/path/to/course/graders/",
+                    }
+                }
+            ]
+        }
+    }
+```
 
 * `test-123`: the name of the queue
 * `SERVER`: XQueue server address
 * `AUTH`: list of username, password
```

Comment:
Nit: It's be nice to be clear for `AUTH` that it's the username/password for a XQueue Django user. Maybe:
`* 'AUTH': List containing [username, password] of XQueue Django user`
I really wish that `AUTH` was more descriptive - but that's beyond this PR scope.